### PR TITLE
completing a phase

### PR DESCRIPTION
**Pull request by JeffreySarnoff**

### CODE REVIEW

**Comment by JeffreySarnoff** [comment]
I resolved the conflicts.
It should merge.

**Comment by JeffreySarnoff** [comment]
here are the resolutions
https://github.com/JuliaMath/BFloat16s.jl/pull/43/conflicts

**Comment by ViralBShah** [comment]
This PR still shows 3 files with conflicts. No merge button.

**Comment by ViralBShah** [comment]
@oscardssmith @maleadt Since you guys have been on a tear - what about this one?

File: src/bfloat16.jl [hunk 1]
```diff
@@ -424,3 +432,10 @@ for F in (:abs, :abs2, :sqrt, :cbrt,
      Base.$F(x::BFloat16) = BFloat16($F(Float32(x)))
   end
 end
+
+Base.atan(y::BFloat16, x::BFloat16) = BFloat16(atan(Float32(y), Float32(x)))
+Base.hypot(x::BFloat16, y::BFloat16) = BFloat16(hypot(Float32(x), Float32(y)))
+Base.hypot(x::BFloat16, y::BFloat16, z::BFloat16) = BFloat16(hypot(Float32(x), Float32(y), Float32(z)))
+Base.clamp(x::BFloat16, lo::BFloat16, hi::BFloat16) = BFloat16(clamp(Float32(x), Float32(lo), Float32(hi)))
```
Comment (maleadt) [inline src/bfloat16.jl:439]: Definitions like this are not valuable, to the contrary actually. The generic definitions work fine already: https://github.com/JuliaLang/julia/blob/48ddd2dc2859f41fc1a82876cc5654fb31d7b2ba/base/intfuncs.jl#L1269-L1272

```julia
julia> clamp(BFloat16(5), BFloat16(5.5), BFloat16(6.5))
BFloat16(5.5)
```

By eagerly demoting to `FLoat32` you're making it impossible for LLVM to emit instructions that would support operations like this, which is now possible after https://github.com/JuliaLang/julia/pull/55486.

Similar arguments can be made for other operations.

**Review by maleadt — CHANGES_REQUESTED** [review]

File: src/bfloat16.jl (hunk 1 above)
Comment (ViralBShah) [reply src/bfloat16.jl:439]: Ok, so this pre-dates the LLVM support. Should we clean up this package and perhaps remove the other operations of this nature?

Comment (JeffreySarnoff) [reply src/bfloat16.jl:439]: yes

Comment (ViralBShah) [reply src/bfloat16.jl:439]: Closing this PR in that case.

**Pull request closed without merging**